Add a note about let's encrypt challenge requiring the port 80 to be accessible (#43070)
quarkusio/quarkus: docs/src/main/asciidoc/tls-registry-reference.adoc

quarkus.management.enabled=true
----

[IMPORTANT]

> .Port 80

Reviewer: ```suggestion
====
.Port 80
```

====

> The Let's Encrypt ACME challenge requires that the application listens on the port `80`.

Reviewer 2: Is it an application itself or the gateway that should be able to accept the challenge over this port ? Indeed, for ex, with NGrok, we have to enable a port 80, but it then forwards it to an application running at for example, localhost:8081. IMHO it is worth making this distinction related to the gateway

Reviewer 2: Oh, I see, you are probably describing the setup in the cloud that you used for the demo ? In that case, should we have a dedicated `NOTE` instead, recomending what to do  in this case re `80`, otherwise, having it as a generic advise may be a bit confusing

Author: It's true everywhere. So, we must have an important admonition. I was hoping to be able to configure it, but it's not possible.

Reviewer 2: Hi @cescoffier I think must be missing something. The Ngrok example which I mentioned above, clearly, I had an endoint running on `8081`, the management interface was also on a non-80 port, sorry for being slow :-)...

Author: that's because ngrok manages the redirect for you.... basically it was opened on port 80.... without you knowing about it.

Reviewer 2: @cescoffier I know that, and this is my point, that was not a Quarkus application problem, it was a gateway accepting port 80 requests, but your note makes it a Quarkus application problem, `the application listens on port 80`, which is not correct.

Now, like I said, I appreciate you did not use NGrok in your demo, and as such this note should be more specific.

Reviewer 2: @cescoffier 
I don't want to be a pain with this review, but right now I'm reading that I have to ensure that effectively my Quarkus endpoint itself must listen on port 80, while with the proxy like NGrok, it is the proxy which listens on 80 and forwards requests to local non-80 ports.
All I'm asking for is for minor clarification making it obvious what part of the system has to listen on 80.

Reviewer 2: By the way, with NGrok you must tell it to allow HTTP 80, just saying...

Reviewer 2: @cescoffier, how about this, can you add, immediately after, `the application listens on port 80`, or when you say it depends on the Internet provider, something like: 

`For example, if you use <<ngrok,NGrok>>, start it with the following command, 'ngrok http --domain <YOUR-NGROK-DOMAIN> 8080 --scheme http'` 

There is an NGrok section at the very end of the doc (which may need an ancor like `ngrok`) explaining that this command must be used...

IMHO, it will just stop the questions like where is this port 80 managed... You are right it is kind of an underlying infrastructure detail ensuring an access to the application over 80, but it is with highlighting it IMHO...

Hope it works for you, thanks

Author: I can change the wording slightly, but this ngork part should only be in the ngrok section. What they do under the hood is not necessary; what's important is that your application is reachable on port 80 (even if you do not realize it). Otherwise, the challenge would not have worked. 

If you have a proxy in front, then this proxy must listen on port 80 and redirect to whatever port your application is. So, I think I will change the wording to not be "listen on port 80" but "reachable on port 80".

> The Let's Encrypt ACME challenge requires that the application listens on the port `80`.

Reviewer: ```suggestion
The Let's Encrypt ACME challenge requires that the application listens on port `80`.
```

> So, make sure the port `80` is accessible from the Internet.

Reviewer: ```suggestion
Ensure the port `80` is accessible from the Internet.
```

> It may require an explicit security policy depending on your hosting provider.

Reviewer: ```suggestion
It might require an explicit security policy, depending on your hosting provider.
```

We also recommend setting `quarkus.http.insecure-requests` to `redirect` to redirect all HTTP requests to HTTPS.

> The ACME challenge accepts self-signed certificate and up to 10 redirection:

Author: No, I'm referring to ACME here.

[source, properties]
----
quarkus.tls.lets-encrypt.enabled=true
quarkus.management.enabled=true

> quarkus.http.insecure-request=redirect

Reviewer 2: ```suggestion
quarkus.http.insecure-requests=redirect
```

Author: good catch!
